beacon-chain/sync/validate_light_client.go

```diff
@@ -0,0 +1,132 @@
+package sync
+
+import (
+	"context"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v5/monitoring/tracing"
+	"github.com/prysmaticlabs/prysm/v5/monitoring/tracing/trace"
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+)
+
+func (s *Service) validateLightClientFinalityUpdate(ctx context.Context, pid peer.ID, msg *pubsub.Message) (pubsub.ValidationResult, error) {
+	// Validation runs on publish (not just subscriptions), so we should approve any message from
+	// ourselves.
+	if pid == s.cfg.p2p.PeerID() {
+		return pubsub.ValidationAccept, nil
+	}
+
+	// TODO keep?
+	// The head state will be too far away to validate any execution change.
+	if s.cfg.initialSync.Syncing() {
+		return pubsub.ValidationIgnore, nil
+	}
+
+	_, span := trace.StartSpan(ctx, "sync.validateLightClientFinalityUpdate")
+	defer span.End()
+
+	log.Info("LC: p2p validateLightClientFinalityUpdate invoked")
+
+	m, err := s.decodePubsubMessage(msg)
+	if err != nil {
+		tracing.AnnotateError(span, err)
+		return pubsub.ValidationReject, err
+	}
+
+	update, ok := m.(interfaces.LightClientFinalityUpdate)
+	if !ok {
+		return pubsub.ValidationReject, errWrongMessage
+	}
+
+	maxActiveParticipants := update.SyncAggregate().SyncCommitteeBits.Len()
+	numActiveParticipants := update.SyncAggregate().SyncCommitteeBits.Count()
+	hasSupermajority := numActiveParticipants*3 >= maxActiveParticipants*2
+
+	last := s.lcStore.LastLCFinalityUpdate
+	if last != nil {
+		// [IGNORE] The finalized_header.beacon.slot is greater than that of all previously forwarded finality_updates,
+		// or it matches the highest previously forwarded slot and also has a sync_aggregate indicating supermajority (> 2/3)
+		// sync committee participation while the previously forwarded finality_update for that slot did not indicate supermajority
+		slot := last.FinalizedHeader().Beacon().Slot
+		lastMaxActiveParticipants := last.SyncAggregate().SyncCommitteeBits.Len()
+		lastNumActiveParticipants := last.SyncAggregate().SyncCommitteeBits.Count()
+		lastHasSupermajority := lastNumActiveParticipants*3 >= lastMaxActiveParticipants*2
+
+		if update.FinalizedHeader().Beacon().Slot < slot {
+			return pubsub.ValidationIgnore, nil
+		}
+		if update.FinalizedHeader().Beacon().Slot == slot && (lastHasSupermajority || !hasSupermajority) {
+			return pubsub.ValidationIgnore, nil
+		}
+	}
+	// [IGNORE] The finality_update is received after the block at signature_slot was given enough time
```

Comment:
we should be testing this

Reply:
The code below does this, no?

Reply:
No i meant with a unit test, but I understand that this PR has no tests so far so its fine unless we want to merge this